keyboards/z34/rules.mk

```diff
@@ -0,0 +1,30 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+# Build Options
+BOOTMAGIC_ENABLE = no      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = yes      # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no      # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = no       # Enable keyboard RGB underglow
+MIDI_ENABLE = no           # MIDI support
+UNICODE_ENABLE = no        # Unicode
+BLUETOOTH_ENABLE = no      # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no          # Audio output on port C6
+FAUXCLICKY_ENABLE = no     # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no        # Enable support for HD44780 based LCDs
+SPLIT_KEYBOARD = yes       # Split common
+LTO_ENABLE = yes           # Enables Link Time Optimization (LTO) which reduces the compiled size
+# There are no extra pins, so we make sure to disable OLED and Encoders
+OLED_ENABLE = no    	   # Enables the use of OLED displays
+ENCODER_ENABLE = no        # Enables the use of encoders
```

Comment:
```suggestion
AUDIO_ENABLE = no          # Audio output
SPLIT_KEYBOARD = yes       # Split common
LTO_ENABLE = yes           # Enables Link Time Optimization (LTO) which reduces the compiled size

# There are no extra pins, so we make sure to disable OLED and Encoders
OLED_SUPPORTED = no        # Enables the use of OLED displays
ENCODER_SUPPORTED = no      # Enables the use of encoders
```

The "supported" directive forcibly disables features. If you want to make sure they're not enabled, this is the way to do it.

Reply:
actually thank YOU @drashna for showing me the supported directive, Just learnt something today. Suggestion has been merged.